src/coreclr/vm/interpexec.cpp

```diff
@@ -934,253 +942,305 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
-                break;
-            case INTOP_LOCALLOC:
-            {
-                size_t len = LOCAL_VAR(ip[2], size_t);
-                void* pMemory = NULL;
+                    // clear the valuetype
+                    memset(vtThis, 0, vtSize);
+                    // pass the address of the valuetype
+                    LOCAL_VAR(callArgsOffset, void*) = vtThis;
 
-                if (len > 0)
+                    ip += 5;
+                    goto CALL_INTERP_SLOT;
+                }
+                case INTOP_ZEROBLK_IMM:
+                    memset(LOCAL_VAR(ip[1], void*), 0, ip[2]);
+                    ip += 3;
+                    break;
+                case INTOP_LOCALLOC:
                 {
-                    pMemory = pThreadContext->frameDataAllocator.Alloc(pFrame, len);
-                    if (pMemory == NULL)
+                    size_t len = LOCAL_VAR(ip[2], size_t);
+                    void* pMemory = NULL;
+
+                    if (len > 0)
                     {
-                        // Interpreter-TODO: OutOfMemoryException
-                        assert(0);
+                        pMemory = pThreadContext->frameDataAllocator.Alloc(pFrame, len);
+                        if (pMemory == NULL)
+                        {
+                            // Interpreter-TODO: OutOfMemoryException
+                            assert(0);
+                        }
+                        if (pMethod->initLocals)
+                        {
+                            memset(pMemory, 0, len);
+                        }
                     }
-                    if (pMethod->initLocals)
+
+                    LOCAL_VAR(ip[1], void*) = pMemory;
+                    ip += 3;
+                    break;
+                }
+                case INTOP_GC_COLLECT:
+                {
+                    // HACK: blocking gc of all generations to enable early stackwalk testing
+                    // Interpreter-TODO: Remove this
                     {
-                        memset(pMemory, 0, len);
+                        pInterpreterFrame->SetTopInterpMethodContextFrame(pFrame);
+                        GCX_COOP();
+                        GCHeapUtilities::GetGCHeap()->GarbageCollect(-1, false, 0x00000002);
                     }
+                    ip++;
+                    break;
                 }
-
-                LOCAL_VAR(ip[1], void*) = pMemory;
-                ip += 3;
-                break;
-            }
-            case INTOP_GC_COLLECT: {
-                // HACK: blocking gc of all generations to enable early stackwalk testing
-                // Interpreter-TODO: Remove this
+                case INTOP_THROW:
                 {
-                    pInterpreterFrame->SetTopInterpMethodContextFrame(pFrame);
-                    GCX_COOP();
-                    GCHeapUtilities::GetGCHeap()->GarbageCollect(-1, false, 0x00000002);
+                    CONTEXT exceptionContext;
```

Comment:
`CONTEXT` is a big structure. Can the CONTEXT variable and the rest of this block be in a noinline helper method so that it does not affect the rest of the method?

Reply:
This is a leftover from the previous way that was using it for resuming here. I am going to remove it and call `DispatchManagedException` without the context argument. That one captures the context inside.